lib/validators/json-schema.js

```diff
@@ -1,321 +1,17 @@
-const { ValidationErrors } = require('./validation-errors');
-
-const SCHEMA_V3 = 'http://json-schema.org/draft-03/schema';
-const SCHEMA_V4 = 'http://json-schema.org/draft-04/schema';
-
-/**
- * @param {Object} schema
- * @returns {[Object, string] | null} Tuple of [schemaMeta, schemaVersion]
- */
-const getSchemaMeta = (schema) => {
-  if (schema && schema.$schema && schema.$schema.includes(SCHEMA_V3)) {
-    return [metaSchemaV3, 'v3'];
-  }
-
-  if (schema && schema.$schema && schema.$schema.includes(SCHEMA_V4)) {
-    return [metaSchemaV4, 'v4'];
-  }
-
-  return null;
-};
-
-/**
- * Returns a proper article for a given string.
- * @param {string} str
- * @returns {string}
- */
-function getArticle(str) {
-  const vowels = ['a', 'e', 'i', 'o', 'u'];
-  return vowels.includes(str.toLowerCase()) ? 'an' : 'a';
-}
-
-const jsonSchemaOptions = {
-  singleError: false,
-  messages: {
-    minLength: (prop, val, validator) =>
-      `The ${prop} property must be at least ${validator} characters long (currently ${val.length} characters long).`,
-    maxLength: (prop, val, validator) =>
-      `The ${prop} property must not exceed ${validator} characters (currently${val.length} characters long).`,
-    length: (prop, val, validator) =>
-      `The ${prop} property must be exactly ${validator} characters long (currently ${val.length} characters long).`,
-    format: (prop, val, validator) =>
-      `The ${prop} property must be ${getArticle(
-        validator[0]
-      )} ${validator} (current value is ${JSON.stringify(val)}).`,
-    type: (prop, val, validator) =>
-      `The ${prop} property must be ${getArticle(
-        validator[0]
-      )} ${validator} (current value is ${JSON.stringify(val)})."`,
-    except: (prop, val) => `The ${prop} property must not be ${val}.`,
-    minimum: (prop, val, validator) =>
-      `The minimum value of the ${prop} must be ${validator} (current value is ${JSON.stringify(
-        val
-      )}).`,
-    maximum: (prop, val, validator) =>
-      `The maximum value of the ${prop} must be ${validator} (current value is ${JSON.stringify(
-        val
-      )}).`,
-    pattern: (prop, val, validator) =>
-      `The ${prop} value (${val}) does not match the ${validator} pattern.`,
-    maxItems: (prop, val, validator) =>
-      `The ${prop} property must not contain more than ${validator} items (currently contains ${val.length} items).`,
-    minItems: (prop, val, validator) =>
-      `The ${prop} property must contain at least ${validator} items (currently contains ${val.length} items).`,
-    divisibleBy: (prop, val, validator) =>
-      `The ${prop} property is not divisible by ${validator} (current value is ${JSON.stringify(
-        val
-      )}).`,
-    uniqueItems: (prop) => `All items in the ${prop} property must be unique.`
-  }
-};
+const { JsonSchemaLegacy } = require('./json-schema-legacy');
+const { JsonSchemaValidator, getSchemaVersion } = require('./json-schema-next');
 
 class JsonSchema {
```

Comment:
A wrapping class to determine which actual validator class to use. A temporary wrapper until we deprecate Amanda and TV4. Overkill: determines a JSON Schema's version _twice_ (once in the wrapper class, another time in the actual validator class).